enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,324 @@
+      control plane nodes be lost.
+
+- If a backup is configured to occur due to `backupReason` or `schedule`:
+  - Given the set of control plane nodes
+  - Given the set of the successful backups recorded in status
+  - Only a single backup pod should be running at a given time. If a backup
+    pod is still running:
+    - If the pod has been running longer than a maximum period (TODO How
+      long?), terminate it and record the backup as having failed.
+    - If the pod has been running for less than the maximum period, defer the
+      scheduled backup for (TODO How long)
+  - If the number of successful backups matches `retentionCount`
+    - Ensure removal of the oldest backup from the node that it was written to
+    - Remove record of removed backup from the api?
+    - Log removal of the backup
+  - Find the set of candidate nodes that have fewer backups than other nodes
+    - Could be 1 node (e.g. 2 nodes have 1 backup and 1 node has zero)
+    - Could be a subset of nodes (e.g. 1 node has 1 backup and 2 nodes have
+      zero)
+    - Could be all nodes (i.e. no successful backups have been taken)
+  - Schedule a backup pod to a random member of the set of candidate nodes
+    - A backup pod should invoke the `cluster-backup.sh` script on the host
+      via chroot and write the backup data to the host.
+    - A backup pod should first check that the available disk space on the
+      node it is running on is a multiple of the size of the node's
+      /var/lib/etcd path. A node needs a minimum amount of available storage
+      to operate reliably and automated backup must make every effort not to
+      exhaust a node's available storage.
+  - When a backup pod terminates, record the succcess or failure to a backup
+    resource
+
+### User Stories
+
+- I want to be alerted after installation or upgrade to the availability of
+  an automated backup capability and be given an indication of the importance
+  of configuring it in a timely manner.
+
+- I want to initiate a cluster backup without requiring a root shell on a
+  control plane node so as to minimize the risk involved.
+
+- I want to schedule recurring cluster backups so that I have recent cluster
+  state to recover with in the event of disaster (i.e. losing 2 control plane
+  nodes).
+
+- I want to have failure to take cluster backups for more than a configurable
+  period to be reported to me via critical alerts.
+
+- I want to minimize the operational impact of taking regular backups.
+
+### Risks and Mitigations
+
+Spreading the spread the cost of automated backups over all control plane
+nodes may not be sufficient to avoid negatively impacting cluster performance
+on IO-constrained hosts. Ideally it would be possible to limit the IO
+overhead of taking a backup via cgroup limits, but it's not clear that this
+would be achievable without the granularity provided by cgroups v2 (not
+slated for GA until RHEL9).
```

Comment:
This is probably my biggest concern, having backup activity adversely affect the performance of the cluster.  One way to indirectly rate limit backups without cgroups v2 might be to constrain the backup's CPU.

Reply:
@hexfusion How effective would it be limit a backup pod's CPU as suggested? I was under the impression that the constraint was almost entirely IO.

Reply:
the perf footprint is:
- CPU/Memory/Disk for leader to serve a snapshot
- wire cost for state travel 
- destination storage medium I/O hit

Reply:
@hexfusion Is it necessary to consider every resource consumed by backup, or is there a subset of them that can be considered 'limiting factors' whose impact it is more important to minimize?